src/frontend/src/binder/expr/binary_op.rs

```diff
@@ -62,4 +64,34 @@ impl Binder {
         )?
         .into())
     }
+
+    /// Bind `||`. Based on the types of the inputs, this can be string concat or others like array
+    /// concat.
```

Comment:
Feels like the job of `infer_type`, just like how `int2 + int8` is mapped to numeric addition while `date + interval` is mapped to temporal/chrono ones.

Reply:
Get it. If we gonna implement `||` and `concat` separately, I'd like to introduce a new expression type of `ConcatOp` and dispatch it in the backend.